react/src/SearchVariants/components/SearchVariantsPage.tsx

```diff
@@ -75,16 +122,35 @@ const SearchVariantsPage: React.FC<SearchVariantsPageProps> = () => {
         }
     };
 
+    const onSuccess = () => {
+        setLoading(false);
+        setSuccess(true);
+    };
+
     const downloadVariantwiseCsv = useDownloadCsv(GET_VARIANTS_SUMMARY_URL, {
         onError,
+        onSuccess,
     });
 
     const downloadParticipantwiseCsv = useDownloadCsv(GET_VARIANTS_BY_PARTICIPANTS_SUMMARY_URL, {
         onError,
+        onSuccess,
     });
 
     const downloadCsv = () => {
         const panel = selectedGenes.map(gene => `ENSG${gene.ensembl_id}`).join(",");
+        const key: QueryKey = [
```

Comment:
This is a nice solution to the problem of a lack of cache-access callbacks! Could you add a brief comment explaining its purpose, in case someone comes to the code later and doesn't remember the specific problem it solves?